EU4toV2/Source/Mappers/CultureMapper/CultureMappingRule.cpp

```diff
@@ -8,30 +10,71 @@ mappers::CultureMappingRule::CultureMappingRule(std::istream& theStream)
 			commonItems::singleString v2Str(theStream);
 			destinationCulture = v2Str.getString();
 		});
-	registerKeyword("eu4", [this](const std::string& unused, std::istream& theStream) 
+	registerKeyword("region", [this](const std::string& unused, std::istream& theStream)
 		{
-			commonItems::singleString eu4Str(theStream);
-			sourceCultures.push_back(eu4Str.getString());
+			commonItems::singleString regionStr(theStream);
+			regions.insert(regionStr.getString());
 		});
-	registerRegex("region|religion|owner|provinceid", [this](const std::string& typeString, std::istream& theStream) 
+	registerKeyword("religion", [this](const std::string& unused, std::istream& theStream)
 		{
-			commonItems::singleString distinguisherStr(theStream);
-		
-			mappers::DISTINGUISHER_TYPES type;
-			if (typeString == "region") { type = DISTINGUISHER_TYPES::REGION; }
-			else if (typeString == "religion") { type = DISTINGUISHER_TYPES::RELIGION; }
-			else if (typeString == "owner") { type = DISTINGUISHER_TYPES::OWNER; }
-			else if (typeString == "provinceid") { type = DISTINGUISHER_TYPES::PROVINCE; }
-
-			distinguishers.insert(std::make_pair(type, distinguisherStr.getString()));
+			commonItems::singleString religionStr(theStream);
+			religions.insert(religionStr.getString());
+		});
+	registerKeyword("owner", [this](const std::string& unused, std::istream& theStream)
+		{
+			commonItems::singleString ownerStr(theStream);
+			owners.insert(ownerStr.getString());
+		});
+	registerKeyword("provinceid", [this](const std::string& unused, std::istream& theStream)
+		{
+			commonItems::singleString provinceStr(theStream);
+			if (stoi(provinceStr.getString())) provinces.insert(stoi(provinceStr.getString()));
```

Comment:
Std::stoi throws an exception on failure, so try/catch instead of checking with an if

Reply:
Thanx, I was under mistaken impression stoi would return 0 for a non-digit string.